Plugin.php

```diff
@@ -105,6 +105,28 @@ public function boot()
             EventRegistry::instance()->setMessageContext($page);
         }, 100);
 
+        /*
+         * replace menu items title with it's translation if available
+         */
+        Event::listen('pages.menu.referencesGenerated', function (&$items) {
+            $iterator = function ($menuItems) use (&$iterator) {
+                $result = [];
+                $locale = App::getLocale();
+                foreach ($menuItems as $item) {
+                    if ($title = array_get($item->viewBag, "localeTitle.$locale")) {
```

Comment:
I'm not sure I'm a fan of property specific translations for the menu use case. Why not have entirely different meta files used between languages, like the repeater logic works right now?

Reply:
@LukeTowers that's how the CMS page fields gets translated currently. I think in that case it's preferable to language specific meta files. What problem do you see exactly?

Reply:
@mjauvin Another thing: Could you separate the condition from the definition again please? :P

Reply:
done @bennothommo

Reply:
I'm not a fan of it for menus specifically because you might have different menu structures, target URLs, and overall menu content in different locales, not just titles.

Reply:
@bennothommo any thoughts on the above?

Reply:
@LukeTowers @mjauvin I don't necessarily have a problem with it being added to the view bag after seeing the YAML below - adding another meta file might be more complexity than it's worth - but I wonder if it could all fall underneath a `locale` property (as an array, perhaps keyed by locale) as opposed to separate properties for each field. This way, it can be extended later with other fields if need be.

Reply:
@bennothommo do you mean a locale property under viewBag or by itself?

Reply:
@mjauvin inside the view bag.

Reply:
Like that @bennothommo ?
```
name: test-menu     
items:
    -   
        title: 'My Title'
        type: url
        url: My URL
        code: ''
        viewBag:
            locale:
                fr: 
                    title: 'Mon Titre'
                    url: 'Mon URL'
                es: 
                    title: 'Mi Titulo'
                    url: 'Mi URL'
            isHidden: '0'
            cssClass: ''
            isExternal: '0'
```

Reply:
done